src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -592,6 +598,40 @@ void GoogleCalendarSyncAdaptor::finalCleanup()
     if (succeededAccounts.size()) {
         setLastSyncSuccessful(succeededAccounts);
     }
+
+    // Delete any events which are not associated with a notebook.
+    // These events are ghost events, caused by a bug which previously
+    // existed in the sync adapter code due to mkcal deleteNotebook semantics.
```

Comment:
In that case, I suppose the reference to deleteNotebook is slightly misleading. Wonder if earlier missing recurrence id support could be related to these ghost events. I recall events with recurrence id but missing main event are currently undeletable from calendar.

I guess this is okayish db clean up, but doing it on every sync seems a bit much. Once per device could be enough already. Not fully objecting this version, though.